File: design-patterns/java/DesignPatterns/src/ar/edu/unlam/cdp/patterns/command/Command.java
package ar.edu.unlam.cdp.patterns.command;

public abstract class Command {

	private String description;
	
	Command(String description) {
		this.description = description;
	}
	
	public abstract void execute();
	public abstract void undo();

	public String getDescription() {
		return description;
	}
	
}

File: design-patterns/cs/Patrones/Patrones/Command/Command.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Patrones.Command
{
    public abstract class Command
    {
        public string Description { get; private set; }

        public Command(string description)
        {
            Description = description;
        }

        #region Metodos Abstractos

        public abstract void Execute();
        public abstract void Undo();

        #endregion
    }
}
